staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -407,7 +410,11 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	err = <-errorChan
 	if err != nil {
 		runtime.HandleError(err)
-		pf.streamConn.Close()
+
+		// If no bytes were sent or received, the connection was never established and streamConn should be closed so that port forwarding can exit.
+		if bytesReceived == 0 && bytesSent == 0 {
```

Comment:
This is good idea. What happens if 1 or more bytes are received or sent but we get [an error](https://github.com/kubernetes/kubernetes/pull/117442/files#diff-03eefd1f0ca3e8aaad80a0a95db19e19d9f092caed73a007645a03363d6676a3R412) because datastream is already closed https://github.com/kubernetes/kubernetes/pull/117442/files#diff-03eefd1f0ca3e8aaad80a0a95db19e19d9f092caed73a007645a03363d6676a3R393